core/templates/services/svg-sanitizer.service.ts

```diff
@@ -211,12 +235,37 @@ export class SvgSanitizerService {
    * trusted. Otherwise returns null.
    */
   getTrustedSvgResourceUrl(base64ImageData: string): SafeResourceUrl | null {
-    if (this.isValidBase64Svg(base64ImageData)) {
+    if (this.isBase64Svg(base64ImageData)) {
+      const sanitizedBase64ImageData = this.removeAllInvalidTagsAndAttributes(
+        base64ImageData);
+
       // eslint-disable-next-line oppia/no-bypass-security-phrase
-      return this.sanitizer.bypassSecurityTrustResourceUrl(base64ImageData);
+      return this.sanitizer.bypassSecurityTrustResourceUrl(
+        sanitizedBase64ImageData);
     }
     return null;
   }
+
+  getIssueURL(
+      invalidTagsAndAttributes: { tags: string[]; attrs: string[] }): string {
+    const baseURL = 'https://github.com/oppia/oppia/issues/new?title=Uploaded%20SVG%20image%20looks%20distorted%20in%20the%20preview&body=The%20image%20file%20is%20attached%20below:%0A%0A%7B%7BIMAGE_HERE%7D%7D%0A%0AScreenshots%20of%20the%20problem:%0A%0A%7B%7BSCREENSHOTS_HERE%7D%7D%0A%0AThe%20invalid%20tags%20and%20attributes%20were:';
+    let updatedURL = baseURL;
+    let invalidTags = invalidTagsAndAttributes.tags;
+    let invalidAttributes = invalidTagsAndAttributes.attrs;
+    if (invalidTags.length) {
+      updatedURL = updatedURL + '%0A%20%20Tags:%20';
+      for (let i = 0; i < Math.min(invalidTags.length, 20); i++) {
```

Comment:
If tag/attr names for a provided image are not made up solely of alphabetic chars/colon, do we catch that? If not, we probably should.

Could you please add some tests for what happens if the tags are invalid, e.g. contain stuff like '%', ':' chars (if that can happen)? We should ensure that (a) a restricted set of chars makes it into invalidTags and invalidAttributes; (b) everything gets encoded safely.

Actually, thinking about it, a better way to do this sort of thing would be to construct the string in plaintext and then use url-interpolation.service.ts to take care of the interpolation, rather than doing it directly like this in an error-prone (and possibly insecure) way.

Reply:
Hi! The current methods in `url-interpolation.service.ts` work to embed the template URL with the values provided as key-value pairs. The `<key>` structures in the template URL get replaced with their corresponding values.

However, in this case, we do not know how many invalid tags and attributes might be interpolated and we would have to first construct a proper template URL and then pass it to the method in `url-interpolation.service.ts`. I think an easier approach would only be to call `encodeURIComponent()` method before appending invalid tags and attributes to the URL. The function itself takes care of problematic chars(like ':', '?') and replaces them with proper URL-encoding.

The method would look something like this:
![Screenshot from 2022-03-09 12-36-36](https://user-images.githubusercontent.com/89375125/157390172-9a63d75f-6450-4152-b209-6c66dcf919b2.png)

Reply:
How about assembling the URL first and then calling encodeURIComponent on the whole thing?

Mistakes happen when you try to assemble strings piecemeal.

Reply:
Sure! I'm on it.

Reply:
Done!